Interop/Tensorflow/src/main/java/org/tribuo/interop/tensorflow/TensorFlowTrainer.java

```diff
@@ -0,0 +1,597 @@
+     * @param epochs The number of SGD epochs to run.
+     * @param testBatchSize The minibatch size to use at test time.
+     * @param loggingInterval The logging interval. Set to -1 to quiesce the loss level logging.
+     * @param checkpointPath The checkpoint path, if using checkpoints.
+     * @param modelFormat The model storage format.
+     */
+    private TensorFlowTrainer(Path graphPath,
+                              GraphDef graphDef,
+                              String outputName,
+                              String initName,
+                              GradientOptimiser optimizer,
+                              Map<String,Float> gradientParams,
+                              FeatureConverter<T> featureConverter,
+                              OutputConverter<T> outputConverter,
+                              int minibatchSize,
+                              int epochs,
+                              int testBatchSize,
+                              int loggingInterval,
+                              Path checkpointPath,
+                              TFModelFormat modelFormat) {
+        if ((graphPath == null) && (graphDef == null)) {
+            throw new IllegalArgumentException("Must supply either a GraphDef or a path to a Graph");
+        }
+        this.graphPath = graphPath;
+        this.graphDef = graphDef;
+        this.outputName = outputName;
+        this.initName = initName;
+        this.optimiserEnum = optimizer;
+        this.gradientParams = Collections.unmodifiableMap(new HashMap<>(gradientParams));
+        this.featureConverter = featureConverter;
+        this.outputConverter = outputConverter;
+        this.minibatchSize = minibatchSize;
+        this.epochs = epochs;
+        this.testBatchSize = testBatchSize;
+        this.loggingInterval = loggingInterval;
+        this.checkpointPath = checkpointPath;
+        this.modelFormat = modelFormat;
+
+        // Validate graph.
+        try (Graph graph = new Graph()) {
+            graph.importGraphDef(graphDef);
+            if (graph.operation(initName) == null) {
+                throw new IllegalArgumentException("Unable to find the initialization operation, expected an op with name '" + initName + "'");
+            }
+            Operation outputOp = graph.operation(outputName);
+            if (outputOp == null) {
+                throw new IllegalArgumentException("Unable to find the output operation, expected an op with name '" + outputName + "'");
+            }
+            Shape outputShape = outputOp.output(0).shape();
+            if (outputShape.numDimensions() != 2) {
+                throw new IllegalArgumentException("Expected a 2 dimensional output, found " + Arrays.toString(outputShape.asArray()));
+            }
+        }
+    }
+
+    /**
+     * Used by the OLCUT configuration system, and should not be called by external code.
+     */
+    @Override
+    public void postConfig() throws IOException {
```

Comment:
It looks like there is graph validation in the non-OLCUT base constructor that is missing here, specifically checking the init and output. Am I missing that happening elsewhere?